crates/polars-core/src/utils/mod.rs

```diff
@@ -554,8 +554,8 @@ macro_rules! df {
 pub fn get_time_units(tu_l: &TimeUnit, tu_r: &TimeUnit) -> TimeUnit {
     use TimeUnit::*;
     match (tu_l, tu_r) {
-        (Nanoseconds, Microseconds) => Microseconds,
-        (_, Milliseconds) => Milliseconds,
+        (_, Nanoseconds) => Nanoseconds,
+        (Milliseconds, Microseconds) => Microseconds,
         _ => *tu_l,
     }
 }
```

Comment:
so this always results in the highest precision between `tu_l` and `tu_r`, right?

```
mn -> n
un -> n
nn -> n
mu -> u
uu -> u
nu -> n
mm -> m
um -> u
nm -> n
```

Reply:
Yes,

9 possible variants (3*3).

3 of them are L=R which is a trivial case.

The variants in which L != R (6):
If Nanoseconds on the right result in Nanoseconds (first match case) = 2 cases
If Nanoseconds on the left result in Nanoseconds (last match case) = 2 cases
ms, us => ms (second match case)
us, ms => us (last match case)

Reply:
9 variants isn't a huge number, maybe this is easier to reason about if you just write them all out?

Reply:
@MarcoGorelli added a new commit to explicitly list all cases. If anyone has a different opinion I can revert that easily